### PR TITLE
Make the signatures of `KokkosBuiltins.h` more general

### PR DESCRIPTION
**Pull request by gojakuch**

### CODE REVIEW

**Comment by gojakuch** [comment]
@kliegeois does that address your today's comment fully?

File: include/clad/Differentiator/KokkosBuiltins.h [hunk 1]
```diff
@@ -108,28 +118,29 @@ inline void deep_copy_pushforward(const View1& dst, const View2& src, T param,
...
-                               const size_t /*d_n6*/, const size_t /*d_n7*/) {
+template <class I, class dI, class View, typename Idx0, typename Idx1,
+          typename Idx2, typename Idx3, typename Idx4, typename Idx5,
+          typename Idx6, typename Idx7>
+inline void
+resize_pushforward(const I& arg, View& v, const Idx0 n0, const Idx1 n1,
+                   const Idx2 n2, const Idx3 n3, const Idx4 n4, const Idx5 n5,
+                   const Idx6 n6, const Idx7 n7, const dI& /*d_arg*/, View& d_v,
+                   const Idx0 /*d_n*/, const Idx1 /*d_n*/, const Idx2 /*d_n*/,
+                   const Idx3 /*d_n*/, const Idx4 /*d_n*/, const Idx5 /*d_n*/,
```
Comment (kliegeois) [inline include/clad/Differentiator/KokkosBuiltins.h:142]: this was fine before as:
```
resize(const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop,
       Kokkos::View<T, P...>& v, const size_t n0 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
       const size_t n1 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
       const size_t n2 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
       const size_t n3 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
       const size_t n4 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
       const size_t n5 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
       const size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
       const size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG) {
  impl_resize(arg_prop, v, n0, n1, n2, n3, n4, n5, n6, n7);
```
But I doubt that this modification hurts.

**Review by kliegeois — COMMENTED** [review]
It looks good and fixed the issue I saw, thanks!
I just have a quick comment.

File: include/clad/Differentiator/KokkosBuiltins.h (hunk 1 above)
Comment (gojakuch) [reply include/clad/Differentiator/KokkosBuiltins.h:142]: yeah, I wasn't sure about that one, but I thought it wouldn't hurt

**Comment by gojakuch** [comment]
@vgvassilev I think we can merge this

**Pull request merged**